lib/iris/_merge.py

```diff
@@ -1228,17 +1227,23 @@ def merge(self, unique=True):
                 data = self._skeletons[group[offset]].data
                 # Ensure the data is represented as a dask array and
                 # slot that array into the stack.
-                if is_lazy_data(data):
+                if iris.util.is_dask_array(data):
```

Comment:
@marqh See this [comment](https://github.com/SciTools/iris/pull/2392/files#r102432818) discussion on usage of `iris.util.is_dask_array` ...